app/controllers/bar-graph.controller.js

```diff
@@ -0,0 +1,93 @@
+(function() {
+  'use strict';
+
+  class BarGraphCtrl {
+    constructor() {
+      // set the dimensions of the canvas
+      this.dimensions = {
+        width: 800,
+        height: 400,
+        margin: {
+          top: 20, 
+          right: 30, 
+          bottom: 30, 
+          left: 80
+        }
+      }
+    }
+    $onChanges(changes) {
+      if (changes.barData) {
+        this.barData = angular.copy(this.barData);
+        if (this.barData.length <= 0) {
+          return;
+        }
+        this.drawBarGraph(this.barData);
+      }
+    }
+    drawBarGraph(barData) {
+      // remove old data
+      d3.select('#bar').select('svg').remove();
+      let axes = this.setAxes(barData);
+      this.drawGraphArea(axes);
+      this.addData(barData, axes);
+    }
+    setAxes(barData) {
+      // set the ranges
+      let axis = {
+        x: d3.scaleBand()
+        .rangeRound([0, this.dimensions.width])
+        .padding(0.5),
+        y: d3.scaleLinear()
+        .rangeRound([this.dimensions.height, 0])
+      }
+      // scale the range of the data
+      axis.x.domain(barData.map((d) => { return d.word; }));
+      axis.y.domain([0, d3.max(barData, (d) => { return d.freq; })]);
+      if (d3.min(barData, (d) => { return d.freq; }) >= 100) {
+        axis.y.domain([100, d3.max(barData, (d) => { return d.freq; })]);
+      }
+      return axis;
+    }
+    drawGraphArea(axis) {
+      // add the SVG element

```

Comment:
jezeli musisz uzywac komentarzy to znaczy ze robisz cos zle. zbierz to w funkcje i uzywaj funkcji. tym bardziej, ze widze, ze wspoldzielisz kod pomiedzy kontrolerami. 

sam dla czystosci zrobilbym pewnie wyrzucenie wszystkich transformacji do jednego serwisu i operacje na nich z poziomu kontrolera z podawaniem odpowiednich wartosci.